src/OpenDebugAD7/AD7DebugSession.cs

```diff
@@ -1,392 +1,379 @@
+            }
+
+            m_logger.Write(LoggingCategory.Telemetry, eventName, propertiesDictionary);
+        }
+
+        private ProtocolException CreateProtocolExceptionAndLogTelemetry(string telemetryEventName, int error, string message)
+        {
+            DebuggerTelemetry.ReportError(telemetryEventName, error);
+            return new ProtocolException(message, new Message(error, message));
+        }
+
+        private bool ValidateProgramPath(ref string program)
+        {
+            // Make sure the slashes go in the correct direction
+            char directorySeparatorChar = Path.DirectorySeparatorChar;
+            char wrongSlashChar = directorySeparatorChar == '\\' ? '/' : '\\';
+
+            if (program.Contains(wrongSlashChar))
             {
-                Stopped(thread);
-                IDebugExceptionEvent2 ee = (IDebugExceptionEvent2)eventObject;
-                string text;
-                ee.GetExceptionDescription(out text);
-                FireStoppedEvent(thread, "exception", text);
-            });
-            RegisterAsyncEventHandler(typeof(IDebugProgramCreateEvent2), (engine, process, program, thread, eventObject) =>
+                program = program.Replace(wrongSlashChar, directorySeparatorChar);
+            }
+
+            program = m_pathConverter.ConvertLaunchPathForVsCode(program);
+            if (!File.Exists(program))
             {
-                Debug.Assert(_program == null, "Multiple program create events?");
-                if (_program == null)
+                // On Windows, check if we are just missing a '.exe' from the file name. This way we can use the same
+                // launch.json on all platforms.
+                if (Utilities.IsWindows())
                 {
-                    _program = program;
-                    _debugProtocolCallbacks.SendLater(new InitializedEvent());
+                    if (!program.EndsWith(".", StringComparison.OrdinalIgnoreCase) && !program.EndsWith(".exe", StringComparison.OrdinalIgnoreCase))
+                    {
+                        string programWithExe = program + ".exe";
+                        if (File.Exists(programWithExe))
+                        {
+                            program = programWithExe;
+                            return true;
+                        }
+                    }
                 }
+                return false;
+            }
+            return true;
+        }
 
-                return _configurationDoneTCS.Task;
-            });
-            RegisterSyncEventHandler(typeof(IDebugProgramDestroyEvent2), (engine, process, program, thread, eventObject) =>
+        private void SetCommonDebugSettings(Dictionary<string, JToken> args, out int sourceFileMappings)
+        {
+            // Save the Just My Code setting. We will set it once the engine is created.
+            m_sessionConfig.JustMyCode = args.GetValueAsBool("justMyCode") ?? m_sessionConfig.JustMyCode;
```

Comment:
The other option is to use `GetValueOrDefault()` if we don't want to do `??` notation